proxy/healthy_endpoints.go

```diff
@@ -11,7 +12,7 @@ type healthyEndpoints struct {
 	endpointRegistry *routing.EndpointRegistry
 }
 
-func (h *healthyEndpoints) filterHealthyEndpoints(endpoints []routing.LBEndpoint, rt *routing.Route) []routing.LBEndpoint {
+func (h *healthyEndpoints) filterHealthyEndpoints(endpoints []routing.LBEndpoint, metrics metrics.Metrics) []routing.LBEndpoint {
```

Comment:
I am a little bit confused about adding `metrics` as an argument here. I am working on another PR in parallel (to add logs), and at the moment I don't see any way to pass logger from `proxy.go` here other than adding logger inside arguments list.